common/app/common/pages_helpers.cljc

```diff
@@ -12,14 +12,67 @@
    [app.common.data :as d]
    [app.common.uuid :as uuid]))
 
+(defn walk-pages
+  "Go through all pages of a file and apply a function to each one"
+  ;; The function receives two parameters (page-id and page), and
+  ;; returns the updated page.
+  [f data]
+  (update data :pages-index #(d/mapm f %)))
+
+(defn select-objects
+  "Get a list of all objects in a page that satisfy a condition"
+  [f page]
+  (filter f (vals (get page :objects))))
+
+(defn update-object-list
+  "Update multiple objects in a page at once"
+  [page objects-list]
+  (update page :objects
+          #(into % (d/index-by :id objects-list))))
+
+(defn get-root-component
+  "Get the root shape linked to the component for this shape, if any"
+  [id objects]
+  (let [obj (get objects id)]
+    (if-let [component-id (:component-id obj)]
+      id
+      (if-let [parent-id (:parent-id obj)]
+        (get-root-component parent-id obj)
+        nil))))
+
 (defn get-children
   "Retrieve all children ids recursively for a given object"
   [id objects]
-  (let [shapes (get-in objects [id :shapes])]
+  (let [shapes (vec (get-in objects [id :shapes]))]
```

Comment:
This `vec` seems redundant, the `:shapes` should be already a vector, if not, some transformation that we apply to it in other side does not respects the type.

Reply:
This is tue. There should be some transformation that does not respect the type, but I wasn't able to find it. Perhaps we can set an issue to fix this.

Reply:
sure, in this case, instead of removing vec, please put a TODO: note about this and I will try to find time to fix todos on the code

Reply:
ok